<a href="https://colab.research.google.com/github/munizgui/CP4_SERS/blob/main/CP4_SERS_EX1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Dataset 1 — Appliances Energy Prediction (UCI)

Fonte: https://archive.ics.uci.edu/dataset/374/appliances+energy+prediction

Situação

Uma empresa de eficiência energética está analisando o comportamento de uma residência de baixo consumo. A equipe deseja identificar períodos de consumo elevado dos eletrodomésticos e observar quais condições de temperatura e umidade estavam presentes nesses momentos.

Etapa A — Orange Data Mining

In [ ]:
import pandas as pd

In [ ]:
dados = pd.read_csv("energydata_SAMPLE_01.csv")

In [ ]:
dados.head

dados.head (10)

,Appliances,lights,T1,RH_1,T2,RH_2,T3,RH_3
0,40,0,20.8900,35.4000,17.7600,39.1633,20.2900,36.9000
1,90,10,21.8900,53.1000,21.2900,45.3600,21.6333,49.2267
2,50,0,21.3900,35.5000,17.6333,40.5300,21.6667,35.2000
3,50,0,21.3900,41.0333,23.8900,34.8400,22.0333,36.9333
4,70,0,19.9633,35.1267,16.4633,40.1267,20.0000,36.4000
5,120,10,22.5000,44.6967,20.8900,40.5900,22.2900,38.2600
6,120,10,19.3900,36.7000,16.8900,39.9000,19.7900,36.6267
7,70,0,22.0000,40.4667,20.6000,40.4000,24.2000,37.3267
8,50,0,21.8900,43.0900,19.5000,44.7300,23.5000,40.7900
9,70,20,20.1000,37.2000,17.7300,39.1633,20.2900,36.5900


In [ ]:
# Dimensões do df1
dados.shape


(1974, 8)

In [ ]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1974 entries, 0 to 1973
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Appliances  1974 non-null   int64  
 1   lights      1974 non-null   int64  
 2   T1          1974 non-null   float64
 3   RH_1        1974 non-null   float64
 4   T2          1974 non-null   float64
 5   RH_2        1974 non-null   float64
 6   T3          1974 non-null   float64
 7   RH_3        1974 non-null   float64
dtypes: float64(6), int64(2)
memory usage: 123.5 KB


In [ ]:
# Estatísticas com Describe
dados.describe()

,Consumo_Eletrodomesticos,lights,Temperatura_1,RH_1,Temperatura_2,RH_2,Temperatura_3,RH_3
count,1974.000000,1974.000000,1974.000000,1974.000000,1974.000000,1974.000000,1974.000000,1974.000000
mean,93.044580,3.839919,21.664703,40.219193,20.293635,40.411386,22.231615,39.209733
std,93.826149,7.937076,1.574883,3.971923,2.126958,4.037686,1.957275,3.288818
min,20.000000,0.000000,16.790000,27.733300,16.100000,24.063300,17.200000,30.133300
25%,50.000000,0.000000,20.700000,37.205825,18.790000,37.760000,20.700000,36.900000
50%,60.000000,0.000000,21.600000,39.590000,20.000000,40.433300,22.100000,38.433300
75%,100.000000,0.000000,22.600000,42.991675,21.500000,43.228325,23.290000,41.741250
max,770.000000,50.000000,26.260000,57.496700,28.917500,56.026700,29.198600,49.226700


In [ ]:
# 2. Renomeie Appliances para Consumo_Eletrodomesticos
dados.rename(columns={'Appliances':'Consumo_Eletrodomesticos',
                    'T1':'Temperatura_1',
                    'T2':'Temperatura_2',
                    'T3':'Temperatura_3'
                    }, inplace=True)

In [ ]:
dados.columns

Index(['Consumo_Eletrodomesticos', 'lights', 'Temperatura_1', 'RH_1',
       'Temperatura_2', 'RH_2', 'Temperatura_3', 'RH_3'],
      dtype='object')

In [ ]:
# 3. Determine o maior consumo de eletrodomésticos registrado na amostra.
dados['Consumo_Eletrodomesticos'].max()

770

In [ ]:
print(f'O maior consumo de eletrodomésticos registrado na amostra foi {dados['Consumo_Eletrodomesticos'].max()} Wh')

O maior consumo de eletrodomésticos registrado na amostra foi 770 Wh


In [ ]:
# 4. Calcule um limiar correspondente a 70% do valor máximo e crie um DataFrame contendo somente os registros acima desse limite.

# Cálculo do valor máximo e limiar utilizando o nome atualizado da coluna
valor_maximo = dados["Consumo_Eletrodomesticos"].max()
limiar = valor_maximo * 0.70

# Filtragem do DataFrame
df_acima_limite = dados[dados["Consumo_Eletrodomesticos"] > limiar]

print("Valor máximo:", valor_maximo)
print("Limiar:", limiar)
print("Quantidade de registros:", len(df_acima_limite))

df_acima_limite

Valor máximo: 770
Limiar: 539.0
Quantidade de registros: 22


,Consumo_Eletrodomesticos,lights,Temperatura_1,RH_1,Temperatura_2,RH_2,Temperatura_3,RH_3
352,620,0,24.3900,44.3333,25.3700,37.7360,26.7300,38.8633
427,650,0,20.8900,36.7000,18.2900,39.6633,21.2000,36.0000
453,560,30,21.3567,36.0900,19.5667,37.1633,21.6333,35.0300
616,570,10,24.8900,32.6667,23.5600,31.1400,23.6000,30.8233
696,590,30,19.8567,47.6633,18.7000,34.1000,18.4267,37.8333
744,740,20,19.5000,42.1333,18.2600,39.1967,19.8900,39.1933
916,750,30,22.4300,43.0000,21.9267,40.1333,21.2300,42.5967
993,690,10,22.5333,45.0267,22.0667,39.7900,22.3900,41.9300
1026,560,0,21.0000,38.9000,19.2900,40.2000,21.7000,37.2000
1173,580,0,21.6000,34.6267,19.5000,33.7600,22.5750,34.8725


In [ ]:
# 5. Conte os registros selecionados e calcule qual percentual da amostra eles representam.

# parte 1. Total de registros na amostra original
total_registros = len(dados)

# parte 2. Contagem dos registros acima do limiar
registros_selecionados = len(df_acima_limite)

# parte3. Cálculo do percentual
percentual = (registros_selecionados / total_registros) * 100

# parte 4. Exibição dos resultados
print(f"Total de registros na amostra: {total_registros}")
print(f"Registros acima do limiar: {registros_selecionados}")
print(f"Percentual da amostra: {percentual:.2f}%")

Total de registros na amostra: 1974
Registros acima do limiar: 22
Percentual da amostra: 1.11%


In [ ]:
# 6. Calcule a temperatura média de T1 e crie outro DataFrame contendo simultaneamente consumo acima de 70% do máximo e temperatura acima da média.

# parte 1. Média da temperatura T1 (Temperatura_1)
media_t1 = dados["Temperatura_1"].mean()

# parte 2. Limiar de 70% do consumo máximo
limiar_consumo = dados["Consumo_Eletrodomesticos"].max() * 0.70

# parte 3. Filtragem com ambas as condições simultâneas
df_consumo_e_temp = dados[
    (dados["Consumo_Eletrodomesticos"] > limiar_consumo) &
    (dados["Temperatura_1"] > media_t1)
]

print(f"Temperatura média T1: {media_t1:.2f} °C")
print(f"Registros encontrados: {len(df_consumo_e_temp)}")

df_consumo_e_temp

Temperatura média T1: 21.66 °C
Registros encontrados: 8


,Consumo_Eletrodomesticos,lights,Temperatura_1,RH_1,Temperatura_2,RH_2,Temperatura_3,RH_3
352,620,0,24.3900,44.3333,25.3700,37.7360,26.7300,38.8633
616,570,10,24.8900,32.6667,23.5600,31.1400,23.6000,30.8233
916,750,30,22.4300,43.0000,21.9267,40.1333,21.2300,42.5967
993,690,10,22.5333,45.0267,22.0667,39.7900,22.3900,41.9300
1182,600,20,24.8900,48.1333,28.9175,37.9475,27.1000,42.4400
1196,580,0,23.6333,47.2267,24.8140,43.4720,26.1667,40.5000
1835,720,10,22.2000,37.9333,21.1000,37.0000,23.1667,36.6267
1964,620,0,22.7000,41.7233,22.4267,39.0000,23.2900,39.2000


# 7. Compare os dois DataFrames e explique o efeito da inclusão da temperatura como segundo critério.

######A adição da temperatura como segundo critério restringiu a análise aos momentos em que o alto consumo coincidiu com ambientes internamente mais quentes.  MétricaDF 1 (Consumo > 70%)DF 2 (Consumo > 70% + Temp1 > Média)Impacto / AlteraçãoQuantidade de Registros228Queda de 14 registros (-63,64%)  Proporção da Amostra Total1,11%0,41%Seleção 2,7 vezes mais restritiva  Consumo Médio630,00 Wh643,75 WhElevação discreta no consumo médio (+13,75 Wh)  Temperatura Média (T1)21,67 °C23,46 °CElevação de +1,79 °C no perfil térmico do grupo  Faixa de Temperatura (T1)19,50 °C a 24,89 °C22,20 °C a 24,89 °CEliminação de todas as ocorrências com $T1 \le 21,66$ °C  Efeitos da Inclusão do Segundo CritérioAumento do Rigor de Filtragem: Ao exigir duas condições simultâneas (operador lógico &), o volume de dados foi reduzido em 63,64% em relação ao primeiro filtro.  Descarte de Picos em Temperaturas Amenas: Foram removidas 14 ocorrências nas quais o consumo era alto (> 539 Wh), mas a temperatura da casa variava apenas entre 19,50 °C e 21,66 °C.  Isolamento de Cenários Específicos: Permite à equipe de eficiência energética investigar se os picos de consumo têm relação direta com o uso de climatização ou resfriamento nos dias de maior calor interno.